In [1]:
import json
with open('/Users/marya/.secret/yelp_api.json') as f: #change the path to match YOUR path!!
    login = json.load(f)
login.keys()

dict_keys(['client-id', 'api-key'])

In [2]:
from yelpapi import YelpAPI
yelp_api = YelpAPI(login['api-key'], timeout_s=5.0)
yelp_api

In [4]:
# Open the jsonfile and display the first level of keys
# open API results WITH json module
import json
with open(r'C:\Users\marya\Downloads\example_yelp_results.json') as f:   #adjust for your path
    json_file = json.load(f)
    
## What type is top-level of json?
type(json_file)


dict

# what are the keys?

In [5]:
json_file.keys()

dict_keys(['businesses', 'total', 'region'])

# what type is region?

In [6]:
type(json_file['region'])


dict

In [7]:
# what keys are in region?
json_file['region'].keys()

dict_keys(['center'])

In [8]:
# what type is center?
type(json_file['region']['center'])

dict

In [9]:
# what are the keys in center?
json_file['region']['center'].keys()

dict_keys(['longitude', 'latitude'])

In [10]:
# Visualize the whole center dict, since lat and long are likely single values
json_file['region']['center']

{'longitude': -76.607666015625, 'latitude': 39.29617825404647}

# Exploring Total

In [11]:
# what is stored under the "total" key?
type(json_file['total'])

int

In [12]:
## what is the value?
json_file['total']

435

there were 435 total results for this Yelp search.

# explore business

In [13]:
# what is in in the businesses key?
type(json_file['businesses'])

list

In [14]:
# what does the first entry of business look like?
json_file['businesses'][0]

{'id': 'D9A33FM394q99o4QtK5YwA',
 'alias': 'faidleys-seafood-baltimore-3',
 'name': 'Faidleys Seafood',
 'image_url': 'https://s3-media3.fl.yelpcdn.com/bphoto/OTjVDCVS7pGopH6GZcfjBA/o.jpg',
 'is_closed': False,
 'url': 'https://www.yelp.com/biz/faidleys-seafood-baltimore-3?adjust_creative=KJtcufKUS887p24u6rjVIQ&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=KJtcufKUS887p24u6rjVIQ',
 'review_count': 1184,
 'categories': [{'alias': 'seafood', 'title': 'Seafood'},
  {'alias': 'beerbar', 'title': 'Beer Bar'}],
 'rating': 4.0,
 'coordinates': {'latitude': 39.291696, 'longitude': -76.62224},
 'transactions': ['delivery'],
 'price': '$$',
 'location': {'address1': '203 N Paca St',
  'address2': '',
  'address3': '',
  'city': 'Baltimore',
  'zip_code': '21201',
  'country': 'US',
  'state': 'MD',
  'display_address': ['203 N Paca St', 'Baltimore, MD 21201']},
 'phone': '+14107274898',
 'display_phone': '(410) 727-4898',
 'distance': 1349.560720156645}

In [15]:
## what are the keys of the first entry in businesses?
json_file['businesses'][0].keys()

dict_keys(['id', 'alias', 'name', 'image_url', 'is_closed', 'url', 'review_count', 'categories', 'rating', 'coordinates', 'transactions', 'price', 'location', 'phone', 'display_phone', 'distance'])

In [17]:
import pandas as pd
biz = pd.DataFrame(json_file['businesses'])
biz.head()

id                                 alias  \
0  D9A33FM394q99o4QtK5YwA          faidleys-seafood-baltimore-3   
1  u65W69AhbjUlvJJBkEhGNQ        miss-shirleys-cafe-baltimore-9   
2  ieS_5zqxDHcWMCm8BKUYbg  thames-street-oyster-house-baltimore   
3  6am8TZAFnvND52MOz-Yctg     mamas-on-the-half-shell-baltimore   
4  ISn7WyGQaFpsVSRSh0NSqg                sal-and-sons-baltimore   

                         name  \
0            Faidleys Seafood   
1         Miss Shirley's Cafe   
2  Thames Street Oyster House   
3    Mama's On The Half Shell   
4                Sal and Sons   

                                           image_url  is_closed  \
0  https://s3-media3.fl.yelpcdn.com/bphoto/OTjVDC...      False   
1  https://s3-media4.fl.yelpcdn.com/bphoto/9FsOyV...      False   
2  https://s3-media1.fl.yelpcdn.com/bphoto/9hGjo5...      False   
3  https://s3-media2.fl.yelpcdn.com/bphoto/HWY8OF...      False   
4  https://s3-media3.fl.yelpcdn.com/bphoto/LmVL4j...      False   

                                                 url  review_count  \
0  https://www.yelp.com/biz/faidleys-seafood-balt...          1184   
1  https://www.yelp.com/biz/miss-shirleys-cafe-ba...          2919   
2  https://www.yelp.com/biz/thames-street-oyster-...          2729   
3  https://www.yelp.com/biz/mamas-on-the-half-she...          1279   
4  https://www.yelp.com/biz/sal-and-sons-baltimor...           153   

                                          categories  rating  \
0  [{'alias': 'seafood', 'title': 'Seafood'}, {'a...     4.0   
1  [{'alias': 'breakfast_brunch', 'title': 'Break...     4.0   
2  [{'alias': 'seafood', 'title': 'Seafood'}, {'a...     4.5   
3  [{'alias': 'bars', 'title': 'Bars'}, {'alias':...     4.0   
4  [{'alias': 'seafood', 'title': 'Seafood'}, {'a...     4.5   

                                         coordinates        transactions  \
0    {'latitude': 39.291696, 'longitude': -76.62224}          [delivery]   
1  {'latitude': 39.2870995, 'longitude': -76.6053...  [delivery, pickup]   
2     {'latitude': 39.28214, 'longitude': -76.59162}          [delivery]   
3   {'latitude': 39.27986, 'longitude': -76.5752399}  [delivery, pickup]   
4       {'latitude': 39.284, 'longitude': -76.59337}          [delivery]   

  price                                           location         phone  \
0    $$  {'address1': '203 N Paca St', 'address2': '', ...  +14107274898   
1    $$  {'address1': '750 E Pratt St', 'address2': '',...  +14105285373   
2   $$$  {'address1': '1728 Thames St', 'address2': '',...  +14434497726   
3    $$  {'address1': '2901 Odonnell St', 'address2': '...  +14102763160   
4     $  {'address1': '1640 Aliceanna St', 'address2': ...  +14106751466   

    display_phone     distance  
0  (410) 727-4898  1349.560720  
1  (410) 528-5373  1028.736468  
2  (443) 449-7726  2090.712792  
3  (410) 276-3160  3328.825798  
4  (410) 675-1466  1817.406978

In [21]:
# add offset to our original api call
search_results = yelp_api.search_query(location='Baltimore, MD,21202',
                                       term='food',
                                       offset = 30)


In [22]:
biz30 = pd.DataFrame(search_results['businesses'])
biz30.head(2)

id                            alias  \
0  Q_SdrcYiJum3OBxTX5HpMQ               monarque-baltimore   
1  vb79aRUD88S9AGqn2Oknaw  homeslyce-mt-vernon-baltimore-3   

                     name                                          image_url  \
0                Monarque  https://s3-media1.fl.yelpcdn.com/bphoto/6CxCuw...   
1  HomeSlyce - Mt. Vernon  https://s3-media2.fl.yelpcdn.com/bphoto/3MqfIl...   

   is_closed                                                url  review_count  \
0      False  https://www.yelp.com/biz/monarque-baltimore?ad...            85   
1      False  https://www.yelp.com/biz/homeslyce-mt-vernon-b...           453   

                                          categories  rating  \
0  [{'alias': 'dinnertheater', 'title': 'Dinner T...     4.0   
1  [{'alias': 'pizza', 'title': 'Pizza'}, {'alias...     4.0   

                                        coordinates        transactions  \
0  {'latitude': 39.284758, 'longitude': -76.599701}          [delivery]   
1  {'latitude': 39.293693, 'longitude': -76.615782}  [pickup, delivery]   

                                            location         phone  \
0  {'address1': '1010 Fleet St', 'address2': None...  +14433841480   
1  {'address1': '336 N Charles St', 'address2': '...  +14435014000   

    display_phone     distance price  
0  (443) 384-1480  1438.609478   NaN  
1  (443) 501-4000   750.670580    $$

In [25]:
biz30.to_json(orient='records')

'[{"id":"Q_SdrcYiJum3OBxTX5HpMQ","alias":"monarque-baltimore","name":"Monarque","image_url":"https:\\/\\/s3-media1.fl.yelpcdn.com\\/bphoto\\/6CxCuwkUInv4poBoqIpQ-A\\/o.jpg","is_closed":false,"url":"https:\\/\\/www.yelp.com\\/biz\\/monarque-baltimore?adjust_creative=U9oMx6DYASGt0l2FwUCaNg&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=U9oMx6DYASGt0l2FwUCaNg","review_count":85,"categories":[{"alias":"dinnertheater","title":"Dinner Theater"},{"alias":"french","title":"French"},{"alias":"steak","title":"Steakhouses"}],"rating":4.0,"coordinates":{"latitude":39.284758,"longitude":-76.599701},"transactions":["delivery"],"location":{"address1":"1010 Fleet St","address2":null,"address3":null,"city":"Baltimore","zip_code":"21202","country":"US","state":"MD","display_address":["1010 Fleet St","Baltimore, MD 21202"]},"phone":"+14433841480","display_phone":"(443) 384-1480","distance":1438.6094784484,"price":null},{"id":"vb79aRUD88S9AGqn2Oknaw","alias":"homeslyce-mt-vernon-bal

In [23]:
businesses = pd.concat([biz, biz30],
                      ignore_index=True)
display(businesses.head(3), businesses.tail(3))


id                                 alias  \
0  D9A33FM394q99o4QtK5YwA          faidleys-seafood-baltimore-3   
1  u65W69AhbjUlvJJBkEhGNQ        miss-shirleys-cafe-baltimore-9   
2  ieS_5zqxDHcWMCm8BKUYbg  thames-street-oyster-house-baltimore   

                         name  \
0            Faidleys Seafood   
1         Miss Shirley's Cafe   
2  Thames Street Oyster House   

                                           image_url  is_closed  \
0  https://s3-media3.fl.yelpcdn.com/bphoto/OTjVDC...      False   
1  https://s3-media4.fl.yelpcdn.com/bphoto/9FsOyV...      False   
2  https://s3-media1.fl.yelpcdn.com/bphoto/9hGjo5...      False   

                                                 url  review_count  \
0  https://www.yelp.com/biz/faidleys-seafood-balt...          1184   
1  https://www.yelp.com/biz/miss-shirleys-cafe-ba...          2919   
2  https://www.yelp.com/biz/thames-street-oyster-...          2729   

                                          categories  rating  \
0  [{'alias': 'seafood', 'title': 'Seafood'}, {'a...     4.0   
1  [{'alias': 'breakfast_brunch', 'title': 'Break...     4.0   
2  [{'alias': 'seafood', 'title': 'Seafood'}, {'a...     4.5   

                                         coordinates        transactions  \
0    {'latitude': 39.291696, 'longitude': -76.62224}          [delivery]   
1  {'latitude': 39.2870995, 'longitude': -76.6053...  [delivery, pickup]   
2     {'latitude': 39.28214, 'longitude': -76.59162}          [delivery]   

  price                                           location         phone  \
0    $$  {'address1': '203 N Paca St', 'address2': '', ...  +14107274898   
1    $$  {'address1': '750 E Pratt St', 'address2': '',...  +14105285373   
2   $$$  {'address1': '1728 Thames St', 'address2': '',...  +14434497726   

    display_phone     distance  
0  (410) 727-4898  1349.560720  
1  (410) 528-5373  1028.736468  
2  (443) 449-7726  2090.712792

id                                              alias  \
37  O7EO2AHxQXRHL1fqodpg7Q  daily-special-authentic-mexican-grill-baltimore-3   
38  ynbOCrd_UK6IyDgEamb4gw                             shoyou-sushi-baltimore   
39  PCU5mBJn44Nfj571TgkhmA                      water-for-chocolate-baltimore   

                                     name  \
37  Daily Special authentic mexican grill   
38                           Shoyou Sushi   
39                    Water for Chocolate   

                                            image_url  is_closed  \
37  https://s3-media2.fl.yelpcdn.com/bphoto/76IbRS...      False   
38  https://s3-media4.fl.yelpcdn.com/bphoto/jFUvyW...      False   
39  https://s3-media1.fl.yelpcdn.com/bphoto/O24M4j...      False   

                                                  url  review_count  \
37  https://www.yelp.com/biz/daily-special-authent...            24   
38  https://www.yelp.com/biz/shoyou-sushi-baltimor...           597   
39  https://www.yelp.com/biz/water-for-chocolate-b...           898   

                                           categories  rating  \
37  [{'alias': 'burgers', 'title': 'Burgers'}, {'a...     4.5   
38  [{'alias': 'sushi', 'title': 'Sushi Bars'}, {'...     4.5   
39  [{'alias': 'breakfast_brunch', 'title': 'Break...     4.5   

                                          coordinates  \
37     {'latitude': 39.29133, 'longitude': -76.61497}   
38  {'latitude': 39.2728666047061, 'longitude': -7...   
39      {'latitude': 39.29032, 'longitude': -76.5905}   

                                  transactions price  \
37  [restaurant_reservation, pickup, delivery]   NaN   
38                                  [delivery]    $$   
39                                    [pickup]    $$   

                                             location         phone  \
37  {'address1': '201 N Charles St', 'address2': N...  +14106895603   
38  {'address1': '1450 Light St', 'address2': '', ...  +14106852989   
39  {'address1': '1841 E Lombard St', 'address2': ...  +14106757778   

     display_phone     distance  
37  (410) 689-5603   818.337595  
38  (410) 685-2989  2618.696592  
39  (410) 675-7778  1611.581406